lib/registries/addon/drafts/draft/metadata/route.ts

```diff
@@ -4,17 +4,28 @@ import RouterService from '@ember/routing/router-service';
 import { inject as service } from '@ember/service';
 import DS from 'ember-data';
 
+import DraftRegistrationModel from 'ember-osf-web/models/draft-registration';
 import Analytics from 'ember-osf-web/services/analytics';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
 
 import { DraftRouteModel } from '../route';
+import MetadataController from './controller';
 
 export default class DraftRegistrationMetadataRoute extends Route {
     @service analytics!: Analytics;
     @service store!: DS.Store;
     @service router!: RouterService;
 
-    model(): DraftRouteModel {
-        return this.modelFor('drafts.draft') as DraftRouteModel;
+    async model(): Promise<DraftRegistrationModel> {
+        const draftRouteModel = this.modelFor('drafts.draft') as DraftRouteModel;
+        const { draftRegistration } = await draftRouteModel.taskInstance;
+        return draftRegistration;
+    }
```

Comment:
I think we want to keep the model as a `DraftRouteModel`. This will allow us, for example, to show a loading indicator while `model.taskInstance.isRunning`.

Reply:
Also, with this, `model` in the controller is no longer the correct type.